frame/elections-phragmen/src/migrations/v6.rs

```diff
@@ -0,0 +1,175 @@
+	deposit: Balance,
+}
+
+pub fn apply<T: Config>() -> Weight {
+	let storage_version = StorageVersion::get::<Pallet<T>>();
+	log::info!(
+		target: "runtime::elections-phragmen",
+		"Running migration for elections-phragmen with storage version {:?}",
+		storage_version,
+	);
+
+	if storage_version <= 5 {
+		let weight = migrate_voting::<T>();
+		weight.saturating_add(migrate_members::<T>());
+		weight.saturating_add(migrate_runners_up::<T>());
+		weight.saturating_add(migrate_candidates::<T>());
+
+		StorageVersion::new(6).put::<Pallet<T>>();
+
+		weight
+	} else {
+		log::warn!(
+			target: "runtime::elections-phragmen",
+			"Attempted to apply migration to V6 but failed because storage version is {:?}",
+			storage_version,
+		);
+		Weight::zero()
+	}
+}
+
+pub fn migrate_voting<T: Config>() -> Weight {
+	let mut translated = 0u64;
+	crate::Voting::<T>::translate::<DeprecatedVoter<T::AccountId, BalanceOf<T>>, _>(|_, voter| {
+		translated.saturating_inc();
+		let bounded_votes: BoundedVec<T::AccountId, ConstU32<{ MAXIMUM_VOTE as u32 }>> =
+			voter.votes.try_into().unwrap();
+		Some(crate::Voter { votes: bounded_votes, stake: voter.stake, deposit: voter.deposit })
+	});
+	T::DbWeight::get().reads_writes(translated, translated)
+}
+
+pub fn migrate_members<T: Config>() -> Weight {
+	let mut translated = 0u64;
+	let _ = crate::Members::<T>::translate::<Vec<SeatHolder<T::AccountId, BalanceOf<T>>>, _>(
+		|maybe_members| {
+			translated.saturating_inc();
+			maybe_members.map(|members| {
+				let bounded_members: BoundedVec<
+					SeatHolder<T::AccountId, BalanceOf<T>>,
+					T::DesiredMembers,
+				> = members.try_into().unwrap();
+				bounded_members
+			})
+		},
+	);
+	T::DbWeight::get().reads_writes(translated, translated)
+}
+
+pub fn migrate_runners_up<T: Config>() -> Weight {
+	let mut translated = 0u64;
+	let _ = crate::RunnersUp::<T>::translate::<Vec<SeatHolder<T::AccountId, BalanceOf<T>>>, _>(
+		|maybe_runners_up| {
+			translated.saturating_inc();
+			maybe_runners_up.map(|runners_up| {
+				let bounded_runners_up: BoundedVec<
+					SeatHolder<T::AccountId, BalanceOf<T>>,
+					T::DesiredRunnersUp,
+				> = runners_up.try_into().unwrap();
+				bounded_runners_up
+			})
+		},
+	);
+	T::DbWeight::get().reads_writes(translated, translated)
+}
+
+pub fn migrate_candidates<T: Config>() -> Weight {
+	let mut translated = 0u64;
+	let _ = crate::Candidates::<T>::translate::<Vec<(T::AccountId, BalanceOf<T>)>, _>(
+		|maybe_candidates| {
+			translated.saturating_inc();
+			maybe_candidates.map(|candidates| {
+				let bounded_candidates: BoundedVec<(T::AccountId, BalanceOf<T>), T::MaxCandidates> =
+					candidates.try_into().unwrap();
+				bounded_candidates
+			})
+		},
+	);
+	T::DbWeight::get().reads_writes(translated, translated)
+}
+
+pub struct MigrateToV6<T>(sp_std::marker::PhantomData<T>);
+impl<T: Config> OnRuntimeUpgrade for MigrateToV6<T> {
+	#[cfg(feature = "try-runtime")]
+	fn pre_upgrade() -> Result<Vec<u8>, &'static str> {
```

Comment:
@gpestana I tested the migrations and they seemed to work as far as I can tell. I didn't manage to think of any clever idea on how to write a check on whether the `Voting` storage map migrated correctly(maybe iterate through all voters and check the length of them, but this may be a bit of a messy solution) but I believe that it should also work correctly.

In the `post_upgrade` I check whether the count of the vectors remained the same(i.e. the original length of the vector was less than the bound) or in case the original vector size was greater than the limit the new vector size should be equal to the limit.